app/policies/task_policy.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+class TaskPolicy < ApplicationPolicy
+  def task
+    @record
+  end
+
+  def user_required?
+    false
+  end
+  private :user_required?
+
+  def show?
+    if @user.present?
+      admin? || task.access_level_public? || task.in_same_group?(@user) || task.author?(@user)
+    else
+      task.access_level_public?
+    end
+  end
+
+  %i[index? new? import_start? import_confirm? import_uuid_check? import_external?].each do |action|
+    define_method(action) { everyone }
+  end
+
+  %i[create?].each do |action|
+    define_method(action) { record_owner? || admin? } # should import_uuid_check and import_external be public?
```

Comment:
> should import_uuid_check and import_external be public?

Previously, it was that way. However, in both actions, we are identifying users with their account link token anyway. So maybe it's just time to reflect that in the policy and perform a "proper" check.

Reply:
Currently, the import is broken, since we `require_user!` in the controller and are not setting the `current_user` accordingly.

Reply:
I skipped require_user! for those actions because we do not want to throw an error in import_uuid_check and we authorize the task creation in import_external anyway. Could you please test if the import works now?

Reply:
Sounds good. However, the check still fails:

<img width="968" alt="Bildschirmfoto 2023-12-18 um 11 29 09" src="https://github.com/openHPI/codeharbor/assets/7300329/d42d8eac-3a3a-418a-85dc-6dc8234532f5">

Now, we still need to change the policy (or, better, configure the `current_user` correctly).

Reply:
It should work now. Could you test again please?

Reply:
Yes, it works now 🎉